BrainPortal/app/models/s3_flat_data_provider.rb

```diff
@@ -448,6 +456,28 @@ def rsync_emulation(src_fileinfos,dst_fileinfos) #:nodoc:
            ]
   end
 
+  # Given a list of FileInfo objects with names like "colname/dir1/dir2/file"
+  # returns just the subset that match any of the file +patterns+,
+  # which are like "dir1/*/f*".
+  def filter_fileinfos_by_patterns(fileinfos, colname, patterns)
+    return fileinfos if patterns.blank?
+    patterns = Array(patterns).map(&:to_s)  # e.g. "dir1/dir2/*"
+    fileinfos.select do |fi|
+      finame  = Pathname.new(fi.name)       # "Collection/dir1/dir2/file"
+      fifirst = finame.each_filename.first  # "Collection"
+      next true  if finame.to_s == colname  # "Collection" == "Collection"
+      next false if fifirst     != colname  # should never happen
+      rest = finame.relative_path_from(colname).to_s  # "dir1/dir2/file"
+      selected = patterns.any? { |pat| File.fnmatch(pat, rest, (File::FNM_PATHNAME | File::FNM_DOTMATCH) ) }
+      #if selected
```

Comment:
Do you want to keep the debug code ?

Reply:
I will remove the debug code once we verified in works correctly.